### PR TITLE
Usage documentation for extensions

### PR DESCRIPTION
**Pull request by jkarsrud**
Changed the documentation of the extensions to be more usage docs than developer docs. Also added a page to /developers with extension-developing guidelines and the roadmap.

### CODE REVIEW

**Comment by jkarsrud** [comment]
Hmm, this got a few more commits than I wanted, but the end result should be good :)

File: developers/extensions.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
+.. extensions:
```
Comment (jodal) [inline developers/extensions.rst]: You need a "_" prefix for "extensions" to be ref-able.

File: developers/extensions.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+.. highlight:: javascript
+
+===================
+Building Extensions
+===================
```
Comment (jodal) [inline developers/extensions.rst:6]: "====" over and under the header is a level higher in the header hierarchy than just below, so you can use "====" below the next headers, and not need to use "^^^^" on the lowest levels. This works nicely as is, just thought to let you know how it works.

File: developers/extensions.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+in order to hook you into the right places, Buster.JS provides a series of
+extension points where you can add custom functionality.
+
+A Buster.JS extension is an object that optionally exposes
+a ``create`` method, that will receive custom configuration, and one
+or more methods that implement an extension hook. The number of hooks is
+expected to increase in the near future.
+
+Create: ``var instance = ext.create([options]);``
+-----------------------------------------
```
Comment (jodal) [inline developers/extensions.rst]: Guess you'll get a warning here about header and line not being equally long.

File: extensions/buster-amd.rst [hunk 4]
```diff
@@ -1,7 +1,66 @@
...
 
-**TODO Port from old site**
+Use an AMD loader to test asynchronous modules. You must provide your own
+loader. By default, a loader that provides ``require(deps, callback)``
+is assumed. This will eventually be pluggable.
+
+Install
+-------
+
+Installation is done using npm: ``npm install buster-amd``.
```
Comment (jodal) [inline extensions/buster-amd.rst]: Would be nice to make the install command a code block, just to make it more visible.

File: extensions/buster-amd.rst [hunk 5]
```diff
@@ -1,7 +1,66 @@
...
+as usual.
+
+Configure
+---------
+
+The AMD extension has one configuration option: a path mapper. The path mapper
+is an optional function that translate Buster.JS paths (which are absolute,
+e.g. ``/test/my-test.js``) to AMD friendly module IDs. The default
+mapper converts ``/test/my-test.js`` to ``test/my-test``, i.e. strips leading
+slash and file suffix. Use this option to e.g. use AMD loader plugins.::
```
Comment (jodal) [inline extensions/buster-amd.rst]: Will ".::" on the end of  the line be rendered as "." or ".:"? If a colon doesn't naturally fit on the end of the sentence, I would put the "::" on a line of its own.

File: extensions/buster-jstestdriver.rst [hunk 6]
```diff
@@ -4,4 +4,29 @@
...
+cases.
+
+Install
+-------
+
+Installation is done using npm: ``npm install buster-jstestdriver``.
+
+Usage
+-----
+Load in your configuration file::
```
Comment (jodal) [inline extensions/buster-jstestdriver.rst:22]: Missing empty line after the header.

File: extensions/buster-lint.rst [hunk 7]
```diff
@@ -1,7 +1,74 @@
...
-**TODO Port from old site**
+Incorporate linting (JsLint or JsHint) in your test runs. Optionally fail test
+runs if lint is found.
+
+Install
+-------
+
+Installation is done using npm: ``npm install buster-lint``.
+
+Usage
```
Comment (jodal) [inline extensions/buster-lint.rst:18]: Should be a header.

File: roadmap.rst [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+breaking bugs have been fixed, and all current extensions have been updated
+to accommodate for API changes etc. ``buster-static`` will likely
+not be included with 1.0, but installable as an add-on.
+
+Buster.JS 1.1
+=============
+
+A revamped ``buster-static`` that can do both Node and Browser tests
+from within the browser is included in the default installation. Installers
+for OSX and Windows.
```
Comment (jodal) [inline roadmap.rst]: s/Node/Node.js/, s/Browser/browser/, s/OSX/OS X/

File: extensions/buster-amd.rst (hunk 5 above)
Comment (jkarsrud) [inline extensions/buster-amd.rst]: It seems it does render ".:", so I put "::" on a separate line.

**Comment by jkarsrud** [comment]
Fixed the issues now, cleaned up the headers as well. Using ==== and ----- does look a lot cleaner. Let me know if there's anything else preventing this from being pulled :)

**Comment by jodal** [comment]
Looks good :-)

**Pull request merged**